File: JavaScript/8js 1/all.js
/*
|--------------------------------------------------------------------------
| Enumerable Magic #1 - True for All?
|--------------------------------------------------------------------------
|
*/

function all(arr, fun) {
   return arr.every(fun);
}


const all = (arr, fun) => arr.every(fun);


function all(arr, fun) {
   for (let i = 0; i < arr.length; i++) {
      if (!fun(arr[i])) return false;
   }
   return true;
}


console.log(
   all([1, 2, 3, 4, 5], function (v) {
      return v < 9;
   }),
   true
);
console.log(
   all([1, 2, 3, 4, 5], function (v) {
      return v > 9;
   }),
   false
);

File: Python/8py 1/all.py
"""
|--------------------------------------------------------------------------
| Enumerable Magic #1 - True for All?
|--------------------------------------------------------------------------
|
"""

def all(seq, fun): 
   return not False in list(map(fun, seq))


def _all(seq, fun): 
   return all(map(fun,seq))


def all(seq, fun): 
   for item in seq: 
      if not fun(item): return False
   return True


print(_all((1, 2, 3, 4, 5), lambda x: x>9), False)
print(_all((1, 2, 3, 4, 5), lambda x: x<9), True)